### PR TITLE
CSI-1027 near integration

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/errors/index.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
 export enum WalletErrorType {
+  INACCESSIBLE_ACCOUNT,
```
Comment (irshadCypherock) [inline src/errors/index.ts]: Error code missing

File: src/near/index.ts [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+  address: string;
+  nearPublicKey: string;
+  node: number;
+  network: string;
+  functionCallGasAmount: string;
+  newAccountAmount: string;
+  coin: NearCoinData;
+
+  constructor(xpub: string, coin: NearCoinData, node = 0) {
+    this.node = node;
```
Comment (irshadCypherock) [inline src/near/index.ts]: Unused variable

File: src/near/index.ts [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+      const changeCount = 1;
+      const changeString = '0000000000000000';
+
+      const gas = intToUintByte(0, 32);
+
+      const decimal = intToUintByte(this.coin.decimal, 8);
+      const contractDummyPadding = intToUintByte(
+        Math.round(gasFees / 10000),
+        16 * 4
+      ); //changing decimal to fit the size only until protobuff
```
Comment (irshadCypherock) [inline src/near/index.ts:141]: Maybe the naming or values are incorrect. Gas is 0, contractPadding is using gasFees.

File: src/near/index.ts [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+      value: string;
+      address: string;
+      isMine: boolean;
+    }>;
+    outputs: Array<{ value: string; address: string; isMine: boolean }>;
+  }> {
+    try {
+      const senderAddress = senderAddressArg || this.address;
+      //@ts-ignore required to convert BigNumber to BN.js instance
+      const bnAmount = Web3.utils.toBN(amount);
```
Comment (irshadCypherock) [inline src/near/index.ts:179]: Why can't we do 

```
      const bnAmount = Web3.utils.toBN(amount.toString());
```

File: src/near/txVerifier.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export default function verifyTxn(signedTxn:string):boolean{
+  return signedTxn.length > 0;
+}
```
Comment (irshadCypherock) [inline src/near/txVerifier.ts]: Does near-api-js have something to verify the signature of a txn?

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: src/near/index.ts (hunk 3 above)
Comment (ghost) [reply src/near/index.ts:141]: this is not an optimal solution but this is the only unused variable big enough to fit near gasfees without changing the structure itself. This is a temp fix till we use protobuff on device side

File: src/near/index.ts (hunk 4 above)
Comment (ghost) [reply src/near/index.ts:179]: I tried this it doesn't works

Comment (ghost) [reply src/near/index.ts:179]: for proper conversion, we need to pass the BN.js object. tostring for large numbers messes up the conversion for example if the amount is 1e+24 the string would be '1e+24' and the converter will treat is as a hex string for some reason

File: src/utils/generateNearAddress.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { base_decode } from 'near-api-js/lib/utils/serialize';
+
+const generateNearAddress = (xpub: string) => {
+  xpub.toUpperCase();
```
Comment (irshadCypherock) [inline src/utils/generateNearAddress.ts]: This doesn't do anything

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

**Review by irshadCypherock — DISMISSED** [review]

**Pull request merged**